accepted/0000-payg-cloud-integration.md

```diff
@@ -0,0 +1,242 @@
+The process for this is not straightforward and adds excessive complexity to users and SUSE teams.
+
+The goal for this RFC is to propose a solution to simplify this process and allow Uyuni to synchronize content directly from RMT cloud servers in a user friendly way.
+
+
+# Detailed design
+[design]: #detailed-design
+
+Our solution will be based on the following steps:
+  - Manage pay-as-you-go ssh connections data on Uyuni
+    - Register pay-as-you-go ssh connections data
+    - Update existing ssh connections data
+    - Delete pay-as-you-go ssh connections data
+  - Develop a new taskomatic task and job schedule to:
+    - Retrieve repositories and authentication data from the pay-as-you-go instance
+    - Register repositories and authentication data on Uyuni
+  - Teach Uyuni reposync how cloud RMT authentication data to synchronize product repositories
+
+
+With this solution the expected user flow would be:
+  - Provide to Uyuni ssh information to connect to the pay-as-you-go instance
+    - uyuni will save this information on the database
+    - start a single job execution to synchronize repositories and cloud RMT authentication data
+    - product loaded from the pay-as-you-go instance will be available to import after the task finishes
+  - Import product using existing "add products" feature (available at UI, API, and cmd)
+    - reposync will be able to download all needed data from cloud RMT server
+  - Bootstrap instances using the existing methods
+
+## Manage pay-as-you-go ssh connection data
+
+User should be able to register and manage ssh connection data to pay-as-you-go instances.
+Connecting to those instances, uyuni should extract repository information and authentication data.
+Users should be able to manage ssh connection data using the web UI or the XML-RPC API.
+
+Supported operations:
+  - Add new ssh connection data for pay-as-you-go
+    - should trigger an execution of the taskomatic task to extract data from the payg-as-you-go instance
+  - Remove ssh connection data
+    - Associated data in `suseCredentials` and `susesccrepositoryauth` should also be removed
+  - Update ssh connection data
+  - List existing ssh connection data
+
+Data to be saved:
+  - machine hostname
+  - ssh port
+  - ssh username
+  - ssh password
+  - ssh client certificate
+  - ssh client certificate password
+  - bastion hostname
+  - bastion ssh port
+  - bastion ssh username
+  - bastion ssh password
+  - bastion ssh client certificate
+  - bastion ssh client certificate password
+
+## Taskomatic task
+
+RMT authentication data have a time to live (TTL). For this reason, we need refresh authentication data from the pay-as-you-go instance periodically.
+For that, a new taskomatic job and schedule need to be developed to:
+  - Connect to pay-as-you-go instance and retrieve the authentication data
```

Comment:
Will the task be fired per-instance or will it iterate through all the known instances?

Reply:
I was thinking of firing the task and go through all the known instances.
However, the task can also be called with a list of instances as parameters to update data from. This can be then used when ssh connection data is added or updated.